modules/agar/src/main/ts/ephox/agar/assertions/ApproxStructures.ts

```diff
@@ -46,6 +46,10 @@ export interface ElementFields {
   children?: StructAssert[];
 }
 
+export interface ElementWithSpecificFields extends Omit<ElementFields, 'classes'> {
+  classes?: string[];
+}
```

Comment:
Sorry correct me if I'm wrong; if we added the `string[]` type to the `ElementFields` interface, we would need to do type assertion, ex: `fields.classes as string[]` or `fields.classes as ArrayAssert[]` and pass it to the class asserting functions is that fine?

Reply:
The problem I have with changing `ElementFields` is it makes it *much* easier to write the wrong thing. The idea of a separate function is to be explicit about the intent: for this structure, I *care* about the classes being only these. If you made `ElementFields` support either, then you get into the situation where if the API is poorly understood, and `str` assertions versus a string is misunderstood, they might just write strings because they assumed that's what they should do, and the compiler didn't complain.